### DISCUSSION TITLE
tcs:routingGroup  Some of the problems

### DISCUSSION THREAD

**Discussion by rogercc116** [openTCS/opentcs#61] [Q&A]
I have two kinds of vehicle in my system，I need to let one pass in part of the area.
I tried to group the routes and set properties for the vehicles (group tcs:routingGroup:test vehicle tcs:routingGroup:test) and set up the configuration file (defaultrouter.shortestpath.edgeEvaluators = DISTANCE,EXPLICIT_PROPERTIES) .
But it doesn't work. Can you teach me how to configure it

**Comment by swltr** [comment]
@rogercc116 Routing groups only work with using `EXPLICIT_PROPERTIES`, as you have to set different cost values for different groups.
You have to leave away `DISTANCE` in your configuration file and configure it to only use `EXPLICIT_PROPERTIES`, and then set the `tcs:routingCostForward<GROUP>` and/or `tcs:routingCostReverse<GROUP>` properties on the paths. (See https://www.opentcs.org/docs/5.6/user/opentcs-users-guide.html#_cost_functions.)
